### ISSUE TITLE
Backward HMM smoother: exp can overflow to inf (0*inf -> NaN)

### ISSUE THREAD

**Issue by docxology** [infer-actively/pymdp#432]
## Summary

In `pymdp/algos.py:871-881` (`_hmm_smoother_scan_row_oriented`), the backward-scan exponential `col_scale = jnp.exp(log_likelihoods[1:] - log_c[1:][:, None])` has no max-shift, so a tiny one-step normalizer `log_c` (a rare observation) makes the exponent large-positive → `exp → inf`, and `transition_mats * col_scale` can then be `inf`, producing `0*inf → NaN` in the backward products.

## Details

```python
col_scale = jnp.exp(log_likelihoods[1:] - log_c[1:][:, None])
M = transition_mats * col_scale[:, None, :]
```

`log_c` is the log of the one-step predictive normalizer (`_log_predictive_normalizer`, `algos.py:712-719`), which is ≤ 0, so `ll - log_c` can be large positive when `log_c` is very negative. The same module's filtering path (`_log_predictive_normalizer` at `:717-719`, and `_condition_on` at `:733-734`) subtracts the row max (`ll_max`) before exponentiating; the smoother does not. Only the final renormalization (`:885`) is clipped, which does not rescue intermediate NaN from `0 * inf`.

Note: any positive per-row rescaling of `M` cancels in the final per-row normalization at `:885`, so the shift is safe to apply.

## Suggested fix

Shift the exponent by the row max, matching the `:717` idiom:

```python
log_col = log_likelihoods[1:] - log_c[1:][:, None]
log_col = log_col - jnp.max(log_col, axis=-1, keepdims=True)
col_scale = jnp.exp(log_col)
```

## Environment

- inferactively-pymdp 1.0.3 (commit 0681ce4)
- Found via static red-team review; pattern cross-checked against the filtering path
